trickle.go

```diff
@@ -0,0 +1,66 @@
+package kapacitor
+
+import (
+	"github.com/influxdata/kapacitor/edge"
+	"github.com/influxdata/kapacitor/models"
+	"github.com/influxdata/kapacitor/pipeline"
+)
+
+func newTrickleNode(et *ExecutingTask, n *pipeline.TrickleNode, d NodeDiagnostic) *TrickleNode {
+	sn := &TrickleNode{
+		node: node{Node: n, et: et, diag: d},
+	}
+	sn.node.runF = sn.runTrickle
+	return sn
+}
+
+type TrickleNode struct {
+	node
+	dims models.Dimensions
+	name string
+}
+
+func (n *TrickleNode) runTrickle(_ []byte) error {
+	consumer := edge.NewConsumerWithReceiver(
+		n.ins[0],
+		n,
+	)
+	return consumer.Consume()
+}
+
+// BeginBatch sets some batch variables on the node, and isn't forwarded.
+func (n *TrickleNode) BeginBatch(b edge.BeginBatchMessage) error {
+	n.dims = b.Dimensions()
+	n.name = b.Name()
+	return nil
+}
+
+// BatchPoint forwards a PointMessage
+func (n *TrickleNode) BatchPoint(bp edge.BatchPointMessage) error {
+	return n.outs[0].Collect(edge.NewPointMessage(
```

Comment:
Same question with `len(outs)`

Reply:
Yes, in the attachment, in the tick code.  if len is not 1, we will have errors way before it gets to here.